pkg/internalapis/cnsvolumeoperationrequest/v1alpha1/cnsvolumeoperationrequest_types.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CnsVolumeOperationRequestSpec defines the desired state of CnsVolumeOperationRequest
+type CnsVolumeOperationRequestSpec struct {
+	// Name represents the name of the instance.
+	// There is no strict naming convention for instances; it is dependent on the caller.
+	Name string `json:"name"`
+}
+
+// CnsVolumeOperationRequestStatus defines the observed state of CnsVolumeOperationRequest
+type CnsVolumeOperationRequestStatus struct {
+	// ID represents the VolumeID or SnapshotID of the backend volume or snapshot.
+	// Populated during successful CreateVolume or CreateSnapshot calls.
+	ID string `json:"id,omitempty"`
+	// Populated with the latest capacity on every successful ExtendVolume call for a volume.
+	Capacity int64 `json:"capacity,omitempty"`
+	// FirstOperationDetails stores the details of the first operation performed on the volume.
+	// For debugging purposes, clients should ensure that this information is never overwritten.
+	// More recent operation details should be stored in the LatestOperationDetails field.
+	FirstOperationDetails OperationDetails `json:"firstOperationDetails,omitempty"`
+	// LatestOperationDetails stores the details of the latest operations performed
+	// on the volume. Should have a maximum of 10 entries.
+	LatestOperationDetails []OperationDetails `json:"latestOperationDetails,omitempty"`
+}
+
+// OperationDetails stores the details of the operation performed on a volume.
+type OperationDetails struct {
+	// TaskInvocationTimestamp represents the time at which the task was invoked.
+	// This timestamp is derived from the cluster and may not correspond to the
+	// task invocation timestamp on CNS.
+	TaskInvocationTimestamp metav1.Time `json:"taskInvocationTimestamp"`
+	// TaskID stores the task for an operation that was invoked on CNS for a volume.
+	TaskID string `json:"taskId"`
+	// OpID stores the OpID for a task that was invoked on CNS for a volume.
+	OpID string `json:"opId,omitempty"`
```

Comment:
When can this be empty?

Reply:
You should get OpId once the task is scheduled in CNS.

Reply:
At CSI, we dont. Govmomi returns the `Task` object that only contains a key. For example:
```
{"log":"2021-05-10T22:24:03.116Z\u0009INFO\u0009volume/manager.go:242\u0009Task: (*object.Task)(0xc000670380)(Task:task-5190)\n","stream":"stderr","time":"2021-05-10T22:24:03.116925172Z"}
```

From there we call `GetTaskInfo` which actually waits for the task result before returning detailed information like opID. 
Heres an example - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/common/cns-lib/volume/manager.go#L243